ecommerce/models.py

```diff
@@ -334,11 +337,55 @@ class FulfilledOrder(Order):
     """An order that has a fulfilled payment"""
 
     @transition(
-        field="state", source=Order.STATE.FULFILLED, target=Order.STATE.REFUNDED
+        field="state",
+        source=Order.STATE.FULFILLED,
+        target=Order.STATE.REFUNDED,
+        custom=dict(admin=False),
     )
-    def refund(self):
-        """Issue a refund"""
-        pass
+    def refund(self, amount: float, reason: str, unenroll_learner: bool):
+        """
+        Records the refund, and optionally attempts to unenroll the learner from
+        the things they bought.
+
+        Args:
+            amount (float): the amount that was refunded
+            reason (str): reason for refunding the order
+            unenroll_learner (bool): also unenroll the learner (if possible)
+        Returns:
+            boolean: if unenroll_learner is set to True and the unenrollment
+            fails, the unenrollment will still happen but this will return False
+            to signify that it failed.
+        """
+        self.transactions.create(
+            data={"reason": reason}, amount=amount, transaction_type="refund"
+        )
+
+        self.state = Order.STATE.REFUNDED
+
+        unenrolls_successful = True
+
+        if unenroll_learner:
+            from courses.api import deactivate_run_enrollment
```

Comment:
why are we importing this here?

Reply:
It was only used here. I moved it to the top in https://github.com/mitodl/mitxonline/pull/480/commits/69b947a8412a5ccaed5ea275414feaa3041371b6